clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -53,4 +53,7 @@
      */
     void sleep(long ms);
 
+    default Timer timer(long timeoutMs) {
```

Comment:
Ack on the first point. For the second, I had initially expected it would be useful to be able to override the method in testing to provide an alternative implementation. I haven't needed it yet, but I'm slightly inclined to keep the option open and avoid the direct dependence on the Timer implementation. I also thought it might make Timer a bit more discoverable since most are familiar with the Time interface.

Reply:
Shall we add javadoc to both of these methods?